napari/_vispy/vendored/image.py

```diff
@@ -288,6 +288,12 @@ def set_data(self, image):
         image : array-like
             The image data.
         """
+        if not self.visible:
```

Comment:
I'd guess we need it for `volume` but I'd like to see a case that fails, and then make the change to fix it.

Not doing important stuff just because you are not visible works for us, but surely vispy doesn't want to do that in general.  That would break things. Maybe there is a better way to get the behavior we want:

1. Not creating the node at all until the data is ready seems like the obvious better way. We used visibility just so we had to change the least code, not because it really makes sense. If we didn't create the new node at all, potentially we could keep drawing the old one until the new one was ready. So we'd get the "draw the stale slice" behavior without a compound visual.

2. Add some new flag/state that blocks `set_data`. Like "frozen" or "disabled" or something. So then it's something we explicitly toggle when it's invisible, rather than making vispy interpret visibility a new way.

Reply:
I'm going to mark this resolved. I have no modified Volume yet, but this will be an incredibly obvious and blatant bug when we start testing more. So I'm not afraid of this suggestion being "lost".

Also on the sprint is this "octree visuals version 0" which I think will solve this also. There will be some type of container visual that will manage its children. The container itself will be always drawn.